pkg/api/testing/unstructured_test.go

```diff
@@ -182,141 +193,159 @@ func TestRoundtripToUnstructured(t *testing.T) {
+	}{
+		{legacyscheme.Scheme, legacyscheme.Codecs},
+		{aggregatorschema.Scheme, aggregatorschema.Codecs},
+		{apiserver.Scheme, apiserver.Codecs},
+	}
+
 	var buf bytes.Buffer
-	for gvk := range legacyscheme.Scheme.AllKnownTypes() {
-		if nonRoundTrippableTypes.Has(gvk.Kind) {
-			continue
-		}
-		if gvk.Version == runtime.APIVersionInternal {
-			continue
-		}
+	for _, api := range apis {
+		for gvk := range api.scheme.AllKnownTypes() {
+			if nonRoundTrippableTypes.Has(gvk.Kind) {
+				continue
+			}
+			if gvk.Version == runtime.APIVersionInternal {
+				continue
+			}
 
-		subtestName := fmt.Sprintf("%s.%s/%s", gvk.Version, gvk.Group, gvk.Kind)
-		if gvk.Group == "" {
-			subtestName = fmt.Sprintf("%s/%s", gvk.Version, gvk.Kind)
-		}
+			subtestName := fmt.Sprintf("%s.%s/%s", gvk.Version, gvk.Group, gvk.Kind)
+			if gvk.Group == "" {
+				subtestName = fmt.Sprintf("%s/%s", gvk.Version, gvk.Kind)
+			}
 
-		t.Run(subtestName, func(t *testing.T) {
-			for reason, gvks := range knownFailureReasons {
-				for _, each := range gvks {
-					if gvk == each {
-						t.Skip(reason)
+			t.Run(subtestName, func(t *testing.T) {
+				for reason, gvks := range knownFailureReasons {
+					for _, each := range gvks {
+						if gvk == each {
+							t.Skip(reason)
+						}
 					}
 				}
-			}
 
-			fuzzer := fuzzer.FuzzerFor(FuzzerFuncs, rand.NewSource(seed), legacyscheme.Codecs)
+				fuzzer := fuzzer.FuzzerFor(FuzzerFuncs, rand.NewSource(seed), api.codecs)
 
-			for i := 0; i < 50; i++ {
-				// We do fuzzing on the internal version of the object, and only then
-				// convert to the external version. This is because custom fuzzing
-				// function are only supported for internal objects.
-				internalObj, err := legacyscheme.Scheme.New(schema.GroupVersion{Group: gvk.Group, Version: runtime.APIVersionInternal}.WithKind(gvk.Kind))
-				if err != nil {
-					t.Fatalf("couldn't create internal object %v: %v", gvk.Kind, err)
-				}
-				fuzzer.Fuzz(internalObj)
+				fuzzzInternal := !slices.Contains(typesWithNoInternalVersions, gvk)
 
-				item, err := legacyscheme.Scheme.New(gvk)
-				if err != nil {
-					t.Fatalf("couldn't create external object %v: %v", gvk.Kind, err)
-				}
-				if err := legacyscheme.Scheme.Convert(internalObj, item, nil); err != nil {
-					t.Fatalf("conversion for %v failed: %v", gvk.Kind, err)
-				}
+				for i := 0; i < 50; i++ {
 
-				// Decoding into Unstructured requires that apiVersion and kind be
-				// serialized, so populate TypeMeta.
-				item.GetObjectKind().SetGroupVersionKind(gvk)
+					item, err := api.scheme.New(gvk)
+					if err != nil {
+						t.Fatalf("couldn't create external object %v: %v", gvk.Kind, err)
+					}
 
-				jsonSerializer := jsonserializer.NewSerializerWithOptions(jsonserializer.DefaultMetaFactory, legacyscheme.Scheme, legacyscheme.Scheme, jsonserializer.SerializerOptions{})
-				cborSerializer := cborserializer.NewSerializer(legacyscheme.Scheme, legacyscheme.Scheme)
+					if fuzzzInternal {
```

Comment:
Nit: IMO, if you adopt my suggestion to put these GVKs in a map, it would then be clearer to remove the variable and directly do the map lookup in the if statement:

```suggestion
					if _, ok := typesWithNoInternalVersions[gvk]; ok {
```